share/completions/bzr.fish

```diff
@@ -1,604 +1,156 @@
-complete -c bzr -l custom --description 'Version info in Custom template-based format'
-complete -c bzr -l python --description 'Version info in Python format'
-complete -c bzr -l rio --description 'Version info in RIO (simple text) format (default)'
-complete -c bzr -l help -s h --description 'Show help message'
-complete -c bzr -l include-file-revisions --description 'Include the last revision for each file'
-complete -c bzr -l include-history --description 'Include the revision-history'
-complete -c bzr -l quiet -s q --description 'Only display errors and warnings'
-complete -c bzr -l revision --description 'ARG, -r See "help revisionspec" for details'
-complete -c bzr -l template --description 'ARG Template for the output'
-complete -c bzr -l usage --description 'Show usage message and options'
-complete -c bzr -l verbose -s v --description 'Display more information'
-complete -c bzr -l template --description 'VERSION_INFO \"Project 1'
-complete -c bzr -l all --description 'Apply list or delete action to all views'
-complete -c bzr -l delete --description 'Delete the view'
-complete -c bzr -l help -s h --description 'Show help message'
-complete -c bzr -l name --description 'ARG Name of the view to define, list or delete'
-complete -c bzr -l quiet -s q --description 'Only display errors and warnings'
-complete -c bzr -l switch --description 'ARG Name of the view to switch to'
-complete -c bzr -l usage --description 'Show usage message and options'
-complete -c bzr -l verbose -s v --description 'Display more information'
-complete -c bzr -l branch --description 'Set identity for the current branch instead of globally'
-complete -c bzr -l directory --description 'ARG, -d Branch to operate on, instead of working directory'
-complete -c bzr -l email --description 'Display email address only'
-complete -c bzr -l help -s h --description 'Show help message'
-complete -c bzr -l quiet -s q --description 'Only display errors and warnings'
-complete -c bzr -l usage --description 'Show usage message and options'
-complete -c bzr -l verbose -s v --description 'Display more information'
+function __fish_bzr_needs_command
+    set cmd (commandline -opc)
+
+    if [ (count $cmd) -eq 1 -a $cmd[1] = 'bzr' ]
+        return 0
+    end
+
+    return 1
+end
+
+function __fish_bzr_using_command
+    set cmd (commandline -opc)
+
+    if [ (count $cmd) -gt 1 ]
+        if [ $argv[1] = $cmd[2] ]
+            return 0
+        end
+    end
+
+    return 1
+end
+
+function __fish_bzr_common_command
+    set cmd (commandline -opc)
+
+    if [ (count $cmd) -gt 1 ]
+        if not contains $cmd[2] help
+            return 0
+        end
+    end
+end
+
+function __fish_bzr_dry_run_command
+    set cmd (commandline -opc)
+
+    if [ (count $cmd) -gt 1 ]
+        if contains $cmd[2] add mv
+            return 0
+        end
+    end
+end
+
+# help commands
+complete -f -c bzr -n '__fish_bzr_needs_command' -a help --description 'Show help.'
```

Comment:
We usually don't use a trailing ".". Brevity is important in descriptions because it allows you to fit more on screen.